homeassistant/util/yaml/loader.py

```diff
@@ -267,6 +267,10 @@ def _load_secret_yaml(secret_path: str) -> JSON_TYPE:
 def secret_yaml(loader: SafeLineLoader, node: yaml.nodes.Node) -> JSON_TYPE:
     """Load secrets and embed it into the configuration YAML."""
     secret_path = os.path.dirname(loader.name)
+    secret_filename = os.path.join(secret_path, SECRET_YAML)
+    if secret_filename == loader.name:
+        _LOGGER.error("secrets.yaml: attempt to load secret from within secrets file")
```

Comment:
Looking at the recursive way secrets work, it might make more sense to *not* throw an error here, but to simply start one level up in the directory tree? 
```
secret_path = os.path.dirname(secret_path)
```

This would allow to load and aggregate secrets from “outside“ the current directory.

Reply:
Not sure. That feels out of scope here, as I'm just trying to fix the core "you shouldn't be doing this particular pattern with secrets" problem, not provide new paths.